nemo/collections/nlp/data/language_modeling/megatron/gpt_dataset.py

```diff
@@ -345,6 +373,9 @@ def __getitem__(self, idx):
         attention_mask, loss_mask, position_ids = _create_ltor_masks_and_position_ids(
             tokens, self.eos_id, self.reset_position_ids, self.reset_attention_mask, self.eod_mask_loss,
         )
+        loss_mask[labels == -1] = 0.0
```

Comment:
You are modifying the loss mask here corresponding to the padded labels, but the attention mask seems unchanged. You can no longer use just a regular `tril` mask here, right?